src/components/Filters/Filters.component.tsx

```diff
@@ -1,53 +1,85 @@
-                            <option value="oldest">Oldest</option>
-                        </select>
+interface FiltersComponentProps {
+    onCategorySelect: (categories: CategoryEnum[]) => void;
+    onSortSelect: (sortOption: SortEnum) => void;
+    onSearch: (searchQuery: string) => void;
+}
+
+export const FiltersComponent: React.FC<FiltersComponentProps> = ({ onCategorySelect, onSortSelect, onSearch }) => {
+    const [selectedCategories, setSelectedCategories] = useState<CategoryEnum[]>([]);
+    const [sortOption, setSortOption] = useState<SortEnum>(SortEnum.PRICE_HIGH_LOW);
+    const [searchQuery, setSearchQuery] = useState<string>('');
+
+    const handleCategorySelect = (category: CategoryEnum) => {
+        const updatedCategories = selectedCategories.includes(category)
+            ? selectedCategories.filter((cat) => cat !== category)
+            : [...selectedCategories, category];
+        setSelectedCategories(updatedCategories);
+        onCategorySelect(updatedCategories);
+    };
+
+    const handleSortSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        const selectedSortOption = event.target.value as SortEnum;
+        setSortOption(selectedSortOption);
+        onSortSelect(selectedSortOption);
+    };
+
+    const handleSearchInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setSearchQuery(event.target.value);
+    };
+
+    const handleSearch = () => {
+        onSearch(searchQuery);
+    };
+
+    return (
+        <>
+            <div className={styles.container}>
+                <section className={styles.filterFlex}>
+                    <div className={styles.searchContainer}>
+                        <input
+                            className={styles.searchBar}
+                            type="search"
+                            placeholder={'Search...'}
+                            value={searchQuery}
+                            onChange={handleSearchInput}
+                        />
+                        <button className={styles.searchButton} onClick={handleSearch}>
+                            <img src={searchIcon} alt="search" />
+                        </button>
+                    </div>
+                    <div className={styles.rightFlex}>
+                        <div className={styles.filterButtonsFlex}>
+                            {Object.values(CategoryEnum).map((category) => (
+                                <button
+                                    key={category}
+                                    className={`${styles.filterButton} ${selectedCategories.includes(category) ? styles.activeFilterButton : ''}`}
+                                    onClick={() => handleCategorySelect(category)}
+                                >
+                                    {category}
+                                </button>
+                            ))}
+                        </div>
+                        <div className={styles.sortFlex}>
```

Comment:
<img width="692" alt="Screenshot 2024-06-18 at 12 14 13" src="https://github.com/igor0806/fe-react-2024/assets/102523692/4b3dd7f0-d7cc-4703-8d7d-29b12f4089cd">
filters don't matched to desing